packages/app/src/components/commons/DeterministicAvatar/index.tsx

```diff
@@ -0,0 +1,242 @@
+import React from "react"
+import Sketch from "react-p5"
+import p5Types from "p5"
+import Random, { genTokenData } from "./utils"
+import { Box } from "@mui/material"
+
+interface DeterministicAvatarProps {
+  publicationId?: string
+  width?: number
+  height?: number
+}
+
+const DeterministicAvatar: React.FC<DeterministicAvatarProps> = ({ publicationId, width = 160, height = 160 }) => {
+  let x: number = 10
+  let y: number = 10
+  let markingsLayers: number = 0
+  let noiseOpacity: number = 1
+  let noiseImg: any
+  let layers: any[] = []
+  let palette: any[] = []
+  const divisions = 20
+
+  let tokenData
+  if (publicationId) {
+    const publicationHash = publicationId?.replace("P-", "").replace(/(\d)-(\d*)/, "$1")
```

Comment:
@auryn-macmillan I'm kind of hacking the publication id here to remove the `P-` and the log index. A couple questions:
1) Do you know if there's a cleaner way to handle this?
2) I'm realizing I actually probably need to somehow include the log index here since the transaction hash could include more than one publication. Based on the [Random algo](https://github.com/gnosis/tabula/pull/179/files#diff-be6a555c735b7bcccbe7dddd5643770a73971f16d704d6ee4b7b6768d9363a52R5) I'm using, do you think it's possible to adapt that to include the log index?

Reply:
If I understand this correctly, you're trying to get a string of hex digits to use as the seed.
Another way you could do this is to just hash the publication ID. That way you get a unique, fixed-length, hex string corresponding to each publication.

Reply:
Oh! Yeah that makes sense. Love it. 👍

Reply:
Actually, yeah. I think it makes sense to just do the hashing in the subgraph.
https://thegraph.com/docs/en/developing/assemblyscript-api/#crypto-api